packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -1,1286 +1,5 @@
-            }
-            if (dartObfuscation == true) {
-                args "-dDartObfuscation=true"
-            }
-            if (dartDefines != null) {
-                args "--DartDefines=${dartDefines}"
-            }
-            if (bundleSkSLPath != null) {
-                args "-dBundleSkSLPath=${bundleSkSLPath}"
-            }
-            if (codeSizeDirectory != null) {
-                args "-dCodeSizeDirectory=${codeSizeDirectory}"
-            }
-            if (extraGenSnapshotOptions != null) {
-                args "--ExtraGenSnapshotOptions=${extraGenSnapshotOptions}"
-            }
-            if (extraFrontEndOptions != null) {
-                args "--ExtraFrontEndOptions=${extraFrontEndOptions}"
-            }
-            args ruleNames
-        }
-    }
-}
-
-class FlutterTask extends BaseFlutterTask {
-    @OutputDirectory
-    File getOutputDirectory() {
-        return intermediateDir
-    }
-
-    @Internal
-    String getAssetsDirectory() {
-        return "${outputDirectory}/flutter_assets"
-    }
-
-    @Internal
-    CopySpec getAssets() {
-        return project.copySpec {
-            from "${intermediateDir}"
-            include "flutter_assets/**" // the working dir and its files
-        }
-    }
-
-    @Internal
-    CopySpec getSnapshots() {
-        return project.copySpec {
-            from "${intermediateDir}"
-
-            if (buildMode == 'release' || buildMode == 'profile') {
-                targetPlatformValues.each {
-                    include "${PLATFORM_ARCH_MAP[targetArch]}/app.so"
-                }
-            }
-        }
-    }
-
-    FileCollection readDependencies(File dependenciesFile, Boolean inputs) {
-      if (dependenciesFile.exists()) {
-        // Dependencies file has Makefile syntax:
-        //   <target> <files>: <source> <files> <separated> <by> <non-escaped space>
-        String depText = dependenciesFile.text
-        // So we split list of files by non-escaped(by backslash) space,
-        def matcher = depText.split(': ')[inputs ? 1 : 0] =~ /(\\ |[^\s])+/
-        // then we replace all escaped spaces with regular spaces
-        def depList = matcher.collect{it[0].replaceAll("\\\\ ", " ")}
-        return project.files(depList)
-      }
-      return project.files();
-    }
-
-    @InputFiles
-    FileCollection getSourceFiles() {
-        FileCollection sources = project.files()
-        for (File depfile in getDependenciesFiles()) {
-          sources += readDependencies(depfile, true)
-        }
-        return sources + project.files('pubspec.yaml')
-    }
-
-    @OutputFiles
-    FileCollection getOutputFiles() {
-        FileCollection sources = project.files()
-        for (File depfile in getDependenciesFiles()) {
-          sources += readDependencies(depfile, false)
-        }
-        return sources
-    }
-
-    @TaskAction
-    void build() {
-        buildBundle()
-    }
-}
+def pathToThisFile = buildscript.sourceFile.parentFile
```

Comment:
I think the reason this exists is to support moving the code without breaking anything.. If I am correct can you add a comment here explaining that.

Reply:
Done.